infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java

```diff
@@ -12,42 +12,42 @@
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 
-import io.fabric8.kubernetes.client.Config;
 import io.fabric8.openshift.client.DefaultOpenShiftClient;
 import io.fabric8.openshift.client.OpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftConfigBuilder;
 import javax.inject.Inject;
 import javax.inject.Named;
 import org.eclipse.che.commons.annotation.Nullable;
 
 /** @author Sergii Leshchenko */
 public class OpenShiftClientFactory {
-  private final Config config;
+  private final OpenShiftConfigBuilder configBuilder;
 
   @Inject
   public OpenShiftClientFactory(
       @Nullable @Named("che.infra.openshift.master_url") String masterUrl,
       @Nullable @Named("che.infra.openshift.username") String username,
       @Nullable @Named("che.infra.openshift.password") String password,
       @Nullable @Named("che.infra.openshift.trust_certs") Boolean doTrustCerts) {
-    config = new Config();
+    configBuilder = new OpenShiftConfigBuilder();
     if (!isNullOrEmpty(masterUrl)) {
-      config.setMasterUrl(masterUrl);
+      configBuilder.withMasterUrl(masterUrl);
     }
 
     if (!isNullOrEmpty(username)) {
-      config.setUsername(username);
+      configBuilder.withUsername(username);
     }
 
     if (!isNullOrEmpty(password)) {
-      config.setPassword(password);
+      configBuilder.withPassword(password);
     }
 
     if (doTrustCerts != null) {
-      config.setTrustCerts(doTrustCerts);
+      configBuilder.withTrustCerts(doTrustCerts);
     }
   }
 
   public OpenShiftClient create() {
-    return new DefaultOpenShiftClient(config);
+    return new DefaultOpenShiftClient(configBuilder.build());
```

Comment:
I'd try to build config once in the constructor, save it in a field and use here.

Reply:
good point